### DISCUSSION TITLE
access order of subject data storage from javascript file

### DISCUSSION THREAD

**Discussion by mingjia1025** [jspsych/jsPsych#1011] [Q&A] [answered]
Thank you for your prompt response to my last question! I just ran into a new issue.  I want to assign one of the three conditions (implemented as a cond variable in Javascript code) to each subject  in a rotating fashion depending on the order in which they completed my online experiment and have their data stored in MySQL database. To achieve this, I get the sense that I need to write some code in my javascript file that enables it to access the number of subject data files already stored in the database, which might be achieved by AJAX call. What exactly should I do for my purpose?

I also don't quite understand the part of the overview about data storage on your official website. It gives me php codes but I don't understand them. I have no experience with web development before so I have no sense of how to set up a webpage running my javascript and connect it to my database. Do you have a video tutorial for that?

**Comment by jodeleeuw** [comment, marked as answer]
Hi @mingjia1025 ,

I don't have any good video tutorials set up for this yet. I think adding one is a great idea, but I'm not sure when I'll have the time to do that.

I do have a JavaScript + PHP library that I use in my own lab for handling this. Here's the [GitHub repository](https://github.com/vassar-cogscilab/php-mysql-save-data).  There are some instructions in the readme file. It might be enough for you to get started, or there might be gaps in the explanation. Feel free to follow up here with questions if you get stuck using it.

I believe that [cognition.run](https://www.cognition.run) now supports subject assignment to conditions, so if you wanted to use a (free!) service instead of setting up your own web server that would be another option.

**Reply by mingjia1025** [reply]
Thank you, Josh! I decided to use the cognition.run for subject assignment. I can run some parts of the javascript for my code there, but there are some new issues with the javascript. My experiment involves the use of some psychophysical stimuli of which I want to change certain parameters across trial, so I searched online and find a set of plug-in functions wrapped in a file 'jspsych-psychophysics.js', which I guess you may know. To access these functions, I need to write some code in javascript that reads in another js file. 

I tried the following:
`const script = document.createElement('script');
script.src = 'jspsych-psychophysics.js';
document.head.appendChild(script);
eval(script)`

but the part of my code involving the psychophysics plug-in still doesn't run. First, I want to make sure the above code did load the functions as I wanted.

**Reply by jodeleeuw** [reply]
I think on cognition.run you can upload the JS file using the menu on the left side of the page:

![image](https://user-images.githubusercontent.com/595524/92426509-1631cd00-f158-11ea-8b5d-51b78f8a9b76.png)

Any JS files that you upload there should be automatically available on the experiment page.

**Reply by mingjia1025** [reply]
so as long as I upload the jspsych-psychophysics.js in the menu, I don't need any code for using the functions in that script. Right?

**Reply by jodeleeuw** [reply]
Yep!
